assignment3/magic_dict.py

```diff
@@ -0,0 +1,7 @@
+
+class Magic_dictionary:
```

Comment:
Magdalena, this week I am your reviewer :) 
I would suggest you to make the major changes, that makes your code easier for the other person to read as well as to check the correctness.
1. Magic_dictionary does not seem to implement the functionality, e.g. no dictionary is stored there. It would be better if make functional, store dictionary and find their prefixes, so that your check is real.
2. Implement unittest for word_search, as well as dictionary (https://docs.python.org/2/library/unittest.html), so that you check that your code works correctly with different inputs as well as take into account wrong input formats. Currently I can not even check if the code works..
3. Please use magic_dictionary class, as it is required in the exercise :)
4. May be it is worth adding docstrings, as I it is a good practice, especially for the other reader
5. Seems like you solved the problem in the reversed way: you went through the list of words and was trying to find each word in the grid. What if the word list is extremely large? May be it is worth just traversing the grid and in case the route has prefix, you can go further or/and find the right word. 
6. Recommend to use "pylint" tool https://www.pylint.org/ to validate the syntax, there are quite some empty lines.. 
Once you have iterated I give more feedback!